src/projects/EnsureThat/EnsureOptions.cs

```diff
@@ -4,15 +4,22 @@ namespace EnsureThat
 {
     public struct EnsureOptions
     {
+        /// <summary>
+        /// If <see cref="CustomExceptionFactory"/> is defined, this exception will be thrown instead of the
+        /// standard exceptions for the particular ensure method.
+        /// Assign using <see cref="WithException(Exception)"/>.
+        /// </summary>
+        public Exception CustomException => CustomExceptionFactory(string.Empty);
```

Comment:
[nit] minor perf hit on method call for folks using this API. As I don't use it, I am not going to be the evangelist that nay-says, but something to keep in mind as the OP _is_ a user of this customization.

Reply:
/I advise we not let performance regressions prevent innovation, I feel an obligation to note it for stakeholders, but we can fix issues when(/IF) they are an issue for someone/

Reply:
Sorry, but could you explain how this has performance hit? Properties are methods under the hood, AFAIK, so this does not change. The factory also returns straight-away the given exception. What am I missing?